include/engine/internal_route_result.hpp

```diff
@@ -27,6 +27,8 @@ struct PathData
     NodeID turn_via_node;
     // name of the street that leads to the turn
     unsigned name_id;
+    // segregated edge-based node that leads to the turn
+    bool is_segregated;
```

Comment:
the fact that we now have `is_segregated` in so many more items makes me think that the maneuver would probably be the best place to put it, since we are pulling through maneuvers anyhow.

Reply:
But we need to pass the flag from the annotatePath to the assembleSteps. Did it with the same logic as name_id, classes, etc.

Reply:
You are right. Right now this is consistent. The internal route-result shouldn't be concerned with this data, but for now it is consistent with the name and we can keep it as is. Since we are setting these kind of values for source/target in assemble-steps anyhow, I would like to remove them from InternalRouteResult at some time.